main.go

```diff
@@ -85,6 +85,24 @@ type Tag struct {
 	SHA     string
 }
 
+func checkForRepCreationRequestBuilder(repo string, branch string, number int, token string) *http.Request {
+	url := fmt.Sprintf("https://api.github.com/repos/%s/contents/templates/%s/%d", repo, branch, number)
```

Comment:
use the URL package here to ensure that the string built from the plugin-in's input parameters is actually valid. We also should make sure that the URL is properly encoded.

If you can validate the input parameters at the start of the plugin that would be cleaner as other functions rely on this data as well. This will reduce the amount of times you need todo similar validation in the other functions in this plug-in